packages/babel-parser/src/plugins/typescript/index.js

```diff
@@ -2930,6 +2932,33 @@ export default (superClass: Class<Parser>): Class<Parser> =>
         return expr;
```

Comment:
Can you move the check before this line? It should simplify `arrow.node` checks and we have `typeParameters` ready. We can leak `isSingleTypeWithoutTrailingComma` out of `tryParse` and throw afterwards.

Reply:
of course